services/core-api/app/api/projects/information_requirements_table/resources/information_requirements_table_list.py

```diff
@@ -1,21 +1,134 @@
+
+    # Only create new requirements when row has filled in required/methods or comments
+    def build_irt_payload_from_excel(self, import_file, project_guid):
+        temp_file = tempfile.NamedTemporaryFile(suffix='.xlsx')
+        temp_file.write(import_file.read())
+        excel_dict = Worksheet()
+        excel_dict.xlsx_to_dict(path=temp_file.file, select_sheet='Sheet1')
+        # Retrieve all valid requirements to cross reference with worksheet "Information" cell content
+        import_errors = []
+        sanitized_irt_requirements = []
+        valid_requirements = Requirements.get_all()
+        valid_requirement_descriptions = [
+            requirement.description.strip().lower() for requirement in valid_requirements
+        ]
+
+        # Start parsing at specific row to avoid metadata in template
+        starting_row_number = 7
+        worksheet_to_parse = excel_dict.sanitize_sheet_items[starting_row_number:]
+        for idx, row in enumerate(worksheet_to_parse):
+            ## Parse requirements.description from "Information" cell in spreadsheet and cross reference with DB source of truth to determine validity
+            information_cell = row.get('Information', '')
+            # Split Information cell on spaces to separate out description from appendix prefix
+            information_cell_split = information_cell.split()
+            # Rejoin words with spaces after removing appendix reference
+            sanitized_information_cell = ' '.join(information_cell_split[1:]).strip().lower()
+            information_cell_is_valid = valid_requirement_descriptions.count(
+                sanitized_information_cell) > 0
+
+            required_cell = self.convert_boolean_string(row.get('Required'))
+            methods_cell = self.convert_boolean_string(row.get('Methods'))
+            comments_cell = row.get('Comments')
+            # Add 2 to offset zero-based "idx" and starting_row_number beginning at table header
+            row_number = idx + starting_row_number + 2
+            # If "Information" cell entry is not valid flag that to user(could have a bad template or added custom rows)
+            if information_cell_is_valid is False:
+                import_errors.append(
+                    f'Row {row_number} - "{" ".join(information_cell_split[1:]).strip()}" is not a valid entry in the Information column.'
+                )
+                continue
+            # If "Required" or "Methods" cell entry is true and "Comments" are 'None' add error
+            if (required_cell is True or methods_cell is True) and comments_cell == 'None':
+                import_errors.append(
+                    f'Row {row_number} - "Required" or "Methods" cells is checked off and requires "Comments" to be provided.'
+                )
+                continue
+
+            is_empty_row = required_cell is False and methods_cell is False and comments_cell == 'None'
+            active_requirement = [
```

Comment:
if we check just for the description, we might fall into the issue with similar descriptions. How to ensure to get the correct requirement_guid for Summary in section 2 instead of Summary in section 5? Would we need to consider the level as well to be able to differentiate between similar descriptions?